### PR TITLE
Refactor creation of heat stack domain

### PR DESCRIPTION
**Pull request by kajinamit**
We have multiple functions but each of these is tiny. Squashing these into a single function saves us from passing multiple instances to each layer.

Also this removes redundant lookup of the global secret.

### CODE REVIEW

File: controllers/heat_controller.go [hunk 1]
```diff
@@ -898,7 +899,15 @@ func (r *HeatReconciler) ensureStackDomain(
 	ctx context.Context,
 	helper *helper.Helper,
 	instance *heatv1beta1.Heat,
+	secret *corev1.Secret,
 ) (ctrl.Result, error) {
+
+	val, ok := secret.Data[instance.Spec.PasswordSelectors.Service]
+	if !ok {
+		return ctrl.Result{}, fmt.Errorf("%s not foind in secret %s", instance.Spec.PasswordSelectors.Service, instance.Spec.Secret)
```
Comment (bshephar) [inline controllers/heat_controller.go:907]: /foind/found/

Comment (kajinamit) [reply controllers/heat_controller.go]: me~~  fixed in the latest version.

**Review by bshephar — APPROVED** [review]

**Pull request merged**